### DISCUSSION TITLE
Typescript issue with Setup With Nextjs Guide

### DISCUSSION THREAD

**Discussion by bllakcn** [pmndrs/zustand#2423] [Q&A]
I want to code along with the official guide on [Setup With Nextjs page](https://arc.net/l/quote/afwtsmtv).
When i get to the part where we create the provider for the context:
```ts
...
export const CounterStoreProvider = ({
  children,
}: CounterStoreProviderProps) => {
  const storeRef = useRef<StoreApi<CounterStore>>()
  if (!storeRef.current) {
    storeRef.current = createCounterStore()
  }

  return (
    <CounterStoreContext.Provider value={storeRef.current}>
      {children}
    </CounterStoreContext.Provider>
  )
}
...
```
i get the following ts and eslint errors:
```
Cannot find namespace 'AuthStoreContext'.

Parsing error: '>' expected.eslint
type AuthStoreContext.Provider = /*unresolved*/ any

Operator '>' cannot be applied to types '{ storeRef: MutableRefObject<StoreApi<AuthStore> | undefined>; "": any; }' and '{ children: ReactNode; }'.ts(2365)
Operator '<' cannot be applied to types 'boolean' and 'RegExp'.ts(2365)
```

This is the case when i export this function as it is shown in the docs. I tried to move this to another client component (which is in this case the `providers.tsx`) the errors vanished. I could not understand what was causing it. Is there anyone that had the same issue?

**Comment by dbritto-dev** [comment]
@bllakcn would you mind creating a minimal repro on stackblitz? I would be happy to help

**Comment by justinherrera** [comment]
I was facing the same issue, until I realized I was using .ts file extension instead of .tsx in provider file

**Reply by dbritto-dev** [reply]
yea, that's why we added a comment with path on each code block
